lib/binance.ex

```diff
@@ -420,6 +420,16 @@ defmodule Binance do
     {:ok, Binance.OrderResponse.new(response)}
   end
 
+  defp parse_order_response({
+         :error,
+         {
+           :binance_error,
+           %{code: -2010, msg: "Account has insufficient balance for requested action."} = reason
+         }
+       }) do
+    {:error, %Binance.InsufficientBalanceError{reason: reason}}
```

Comment:
Actually looking at it now, It might be better to sub-package this to something like `%Binance.Error.InsufficientBalance{}` and use `%Binance.Error{}` for all other errors that are not implements as their own struct yet. 

That and 'InsufficientBalance' already implies that the account doesn't have enough balance for the action so enforcing a `:reason` key is probably not needed too

But we can refactor that at a later point